File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    if(head==NULL) 
    {
        return head ;
    }

    struct ListNode *temp ;
    temp = head ;
    int count = 0 ;
    while(temp!=NULL)
    {
        temp=temp->next ;
        count++ ;
    }
    if(n == count)
    {
        struct ListNode *del = head;
        head = head->next;
        free(del);
        return head;
    }
    temp=head ;
    int x=count-n-1 ;
    int y=0 ;
    while(temp!=NULL) 
    {
        if(y==x)
        {
            struct ListNode *delt ;
            delt=temp->next ;
            temp->next=delt->next ;
            free(delt) ;
            return head ;
        }
        y++ ;
        temp=temp->next ;
    } return head ;
}

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
    int i=1;
  ListNode temp=head;
    ListNode ptr=head;
    while(i<=n){
        ptr=ptr.next;
        i++;
    }

    if(ptr==null){
        return head.next;
    }
   ListNode prev=head;
  while(ptr!=null){
    prev=head;
    head=head.next;
    ptr=ptr.next;
  }
  prev.next=head.next;
  return temp;



    }
}
       